modules/swagger-codegen/src/main/resources/lua/api.mustache

```diff
@@ -0,0 +1,141 @@
+}
+
+local function new_{{classname}}(host, basePath, schemes)
+  local schemes_map = {}
+  for _,v in ipairs(schemes) do
+    schemes_map[v] = v
+  end
+  local default_scheme = schemes_map.https or schemes_map.http
+  return setmetatable({
+    host = host;
+    basePath = basePath or "{{{basePath}}}";
+    schemes = schemes_map;
+    default_scheme = default_scheme;
+    http_username = nil;
+    http_password = nil;
+    api_key = {};
+    access_token = nil;
+  }, {{{packageName}}}_mt)
+end
+
+{{#operation}}
+function {{classname}}:{{operationId}}({{#allParams}}{{paramName}}{{#hasMore}}, {{/hasMore}}{{/allParams}})
+    local req = http_request.new_from_uri({
+        scheme = self.default_scheme;
+        host = self.host;
+        path = string.format("%s{{{vendorExtensions.x-codegen-path}}}{{#queryParams}}{{#-first}}?{{/-first}}{{{baseName}}}=%s{{^-last}}&{{/-last}}{{/queryParams}}",
+            self.basePath{{#pathParams}}, {{paramName}}{{/pathParams}}{{#queryParams}}, http_util.encodeURIComponent({{paramName}}){{/queryParams}});
+    })
+
+    -- set HTTP verb
+    req.headers:upsert(":method", "{{httpMethod}}")
+
+    {{#hasConsumes}}
+    -- TODO: create a function to select proper accept
+    -- ref: https://github.com/swagger-api/swagger-codegen/pull/6252#issuecomment-321199879
+    --local var_content_type = { {{#consumes}}"{{{mediaType}}}"{{#hasMore}}, {{/hasMore}}{{/consumes}} }
+    req.headers:upsert("accept", {{#consumes}}{{#-first}}"{{{mediaType}}}"{{/-first}}{{/consumes}})
+
+    {{/hasConsumes}}
+    {{#hasProduces}}
+    -- TODO: create a function to select proper content-type
+    -- ref: https://github.com/swagger-api/swagger-codegen/pull/6252#issuecomment-321199879
+    --local var_accept = { {{#produces}}"{{{mediaType}}}"{{#hasMore}}, {{/hasMore}}{{/produces}} }
+    req.headers:upsert("content-type", {{#produces}}{{#-first}}"{{{mediaType}}}"{{/-first}}{{/produces}})
+
+    {{/hasProduces}}
+    {{#headerParams}}
+    req.headers:upsert("{{baseName}}", {{paramName}})
+    {{/headerParams}}
+
+    {{#formParams}}
+    {{#-first}}
+    req:set_body(http_util.dict_to_query({
+    {{/-first}}
+        ["{{baseName}}"] = {{paramName}};
+    {{#-last}}
+    }))
+    {{/-last}}
+    {{/formParams}}
+    {{#bodyParams}}
+    req:set_body(dkjson.encode({{paramName}}))
+
+    {{/bodyParams}}
+    {{#authMethods}}
+    {{#isApiKey}}
+    {{#isKeyInHeader}}
+    -- api key in headers '{{keyParamName}}'
+    req.headers:upsert("{{{name}}}", api_key['{{{keyParamName}}}'])
+    {{/isKeyInHeader}}
+    {{#isKeyInQuery}}
+    -- TODO: api key in query '{{keyParamName}}'
+    {{/isKeyInQuery}}
+    {{/isApiKey}}
+    {{#isBasic}}
+    -- HTTP basic auth
+    req.readers:upsert("authorization", "Basic " .. basexx.to_base64(self.http_username .. " " .. self.http_password))
+    {{/isBasic}}
+    {{#isOAuth}}
+    -- oAuth
+    req.headers:upsert("authorization", "Bearer " .. self.access_token)
+    {{/isOAuth}}
+    {{/authMethods}}
+
```

Comment:
It shouldn't. It's a feature in mustache template library that the following will not result in any line or whitespace
```
    {{#thisIsFalse}}
    something here...
    {{/thisIsFalse}}
```